specification/v2_ux.md

```diff
@@ -0,0 +1,1079 @@
+- **Resizable panels** - Allow users to adjust panel sizes on feature screens
+
+## Navigation Model
+
+```
+┌─────────────────────────────────────────────────────────────────────────────────────────┐
+│  [Server Name ▼]  ● Connected (23ms)  [Tools] [Resources] [Prompts] [Logs] [Tasks] [History] [Disconnect] │
+├─────────────────────────────────────────────────────────────────────────────────────────┤
+│                                                                                         │
+│                                   Full-width content                                    │
+│                                                                                         │
+└─────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+- **Menu bar** displays server name, connection status with latency, and disconnect button when connected
+- **Top navigation** links to Tools, Resources, Prompts, Logs, Tasks, and History sections
+- **Connection indicator** shows latency from periodic `ping` requests
+- **No sidebar** - content areas use full width
+
+## Screen Flows
+
+### Server List (Home)
+
+Initial screen shown when no server is connected. Displays server cards in a responsive grid (1-2 columns).
+
+```
+┌─────────────────────────────────────────────────────────┐
+│                    MCP Inspector                        │
+│                                        [+ Add Server ▼] │
+├─────────────────────────────────────────────────────────┤
+│                                                         │
+│  ┌─────────────────────────┐  ┌─────────────────────────┐
+│  │ Server Card 1           │  │ Server Card 2           │
+│  │ (see below)             │  │ (see below)             │
+│  └─────────────────────────┘  └─────────────────────────┘
+│                                                         │
+│  ┌─────────────────────────┐                            │
+│  │ Server Card 3           │                            │
+│  └─────────────────────────┘                            │
+│                                                         │
+└─────────────────────────────────────────────────────────┘
+```
+
+**Add Server Options:**
+- **Add manually** - Opens form modal for manual configuration
+- **Import config** - Paste/upload internal `mcp.json` config
+- **Import server.json** - Import MCP Registry format ([Issue #922](https://github.com/modelcontextprotocol/inspector/issues/922))
+
+**Functionality:**
+- List servers from `mcp.json` config file
+- Add/edit/delete servers via CRUD endpoints on proxy
+- Responsive grid layout (1 column mobile, 2 columns desktop)
+
+### Server Connection Card
+
+Each server in the list is displayed as a card with connection controls and status.
+
+```
+┌─────────────────────────────────────────────────────────┐
+│ [Icon] Server Name              v1.0.0                  │
+│        STDIO                    [●] Connected  [Toggle] │
```

Comment:
There should be a [OAuth Debug] button that would open a modal or go to a screen for stepping through the oauth flow.